java-connectors/kafka-connect-azure-eventhubs/src/test/java/io/lenses/java/streamreactor/connect/azure/eventhubs/source/BlockingQueueProducerProviderTest.java

```diff
@@ -74,17 +77,25 @@ void whenConstructorInvokedWithParameters_ThenMockKafkaConsumerShouldBeCreatedAn
         mockedOffsetProvider);
     when(azureConfigMock.getString(AzureEventHubsConfigConstants.CONSUMER_OFFSET)).thenReturn(
         earliestOffset);
+    KeyValueTypes mockedKeyValueTypes = mock(KeyValueTypes.class);
```

Comment:
This test is a little tricky to read.   Just a wall of code, different mock objects mixed together.  It's hard to see what's going on.

I bet you didn't even realise you had this line duplicated in all the craziness :

`    when(mockedKeyValueTypes.getKeyType()).thenReturn(mockedType);`


Try adding a little line spacing, to separate out different mocks being defined.

And try grouping the mock definition `(Blah blahMock = mock...)` with the mock behaviour `(when(blahMock)...)`

```
@Test
  void shouldCreateMockKafkaConsumerAndLogWhenConstructorInvokedWithParameters() {
    // Given
    final String earliestOffset = "earliest";

    final AzureEventHubsConfig azureConfigMock = mock(AzureEventHubsConfig.class);
    when(azureConfigMock.getString(AzureEventHubsConfigConstants.CONSUMER_OFFSET)).thenReturn(earliestOffset);

    final SourceDataType mockedType = mock(SourceDataType.class);
    final Class stringDeserializerClass = StringDeserializer.class;
    when(mockedType.getDeserializerClass()).thenReturn(stringDeserializerClass);

    final KeyValueTypes mockedKeyValueTypes = mock(KeyValueTypes.class);
    when(mockedKeyValueTypes.getKeyType()).thenReturn(mockedType);
    when(mockedKeyValueTypes.getValueType()).thenReturn(mockedType);

    final BlockingQueueProducerProvider testObj = new BlockingQueueProducerProvider(mock(TopicPartitionOffsetProvider.class));

    // When
    BlockingQueuedKafkaProducer consumer;
    try (MockedConstruction<KafkaConsumer> ignored = Mockito.mockConstruction(KafkaConsumer.class)) {
      consumer = testObj.createProducer(azureConfigMock, new ArrayBlockingQueue<>(1), mockedKeyValueTypes);
    }

    // Then
    verify(azureConfigMock).getString(AzureEventHubsConfigConstants.CONNECTOR_NAME);
    verify(azureConfigMock).getString(getPrefixedKafkaConsumerConfigKey(GROUP_ID_CONFIG));
    verify(mockedType, times(2)).getDeserializerClass();
    assertNotNull(consumer);
    assertEquals(1, logWatcher.list.size());
    assertTrue(logWatcher.list.get(0).getFormattedMessage().startsWith(LOG_MESSAGE_PREFIX));
  }
```


After that I would try extracting out methods to make the test a little more readable so it fits on one screen..

Reply:
Yeah, I changed this code a bit, cause it didn't make sense to create this in Task